src/sagemaker_containers/c/libchangehostname.c

```diff
@@ -12,23 +12,81 @@
 // language governing permissions and limitations under the License.
 
 #include <Python.h>
+#include <stdio.h>
 #include <stdlib.h>
+#include "jsmn.h"
 
+static int jsoneq(const char *json, jsmntok_t *tok, const char *s)
+{
+    if (tok->type == JSMN_STRING && (int)strlen(s) == tok->end - tok->start &&
+        strncmp(json + tok->start, s, tok->end - tok->start) == 0)
+    {
+        return 0;
+    }
+    return -1;
+}
 
 int libchangehostname(char *name, size_t len)
 {
-  const char *val = getenv("SM_CURRENT_HOST");
+    int r;
+    FILE *file = fopen("/opt/ml/input/config/resourceconfig.json", "r");
 
-  strncpy(name, val, len);
-  return 0;
-}
+    fseek(file, 0, SEEK_END);
+
+    long length = ftell(file);
+    fseek(file, 0, SEEK_SET);
+
+    char * json_string = malloc(length);
+    fread(json_string, 1, length, file);
+    json_string[length] = '\0';
+
+    fclose(file);
+
+    jsmn_parser parser;
+    jsmntok_t token[1024];
+
+
+    jsmn_init(&parser);
+    r = jsmn_parse(&parser, json_string, strlen(json_string), token, sizeof(token) / sizeof(token[0]));
+
+
+	if (r < 0) {
+		printf("Failed to parse JSON: %d\n", r);
+		return 1;
+	}
 
+	/* Assume the top-level element is an object */
+	if (r < 1 || token[0].type != JSMN_OBJECT) {
+		printf("Object expected\n");
+		return 1;
+	}
 
-static PyObject* libchangehostname_call(PyObject* self, PyObject* args) {
+    /* Loop over all keys of the root object */
+    int i = 1;
+    for (; i < r; i++)
```

Comment:
why not combine ll. 65-66?

Reply:
Unfortunately, it is against C99 conventions to initialize a variable inside a for loop.

Reply:
ah, that's right. I think I'm used to seeing it as:

```c
int i;
for (i = 1; i < r; i++)
```

but I don't have a strong opinion about it